driver-sync/src/test/functional/com/mongodb/client/AbstractClientEncryptionRewrapManyDataKeyProseTest.java

```diff
@@ -147,4 +159,26 @@ public void rewrapWithSeparateClientEncryption(final String srcProvider, final S
         assertEquals(testString, clientEncryption1.decrypt(ciphertext));
         assertEquals(testString, clientEncryption2.decrypt(ciphertext));
     }
+
+    @Test
+    public void shouldThrowClientErrorWhenProviderIsNotSpecified() {
+        //given
+        String expectedErrorMessage = "Missing the provider but supplied a master key in the RewrapManyDataKeyOptions";
+        BsonDocument dstKey = MASTER_KEYS_BY_PROVIDER.get("aws");
```

Comment:
The [prose test](https://github.com/mongodb/specifications/commit/4703757119d96fa1438158ece0a30c2186f47800) says "masterKey: {}", but here it seems to be getting the aws key - is this intentional?

Reply:
This test checks the expected behavior when any key is present, so I used a more descriptive dataset that was already available in the class. However, to match the spec more accurately, I updated it to 'masterKey: {}'. Good catch!